server/src/main/java/org/opensearch/rest/action/admin/cluster/RestWlmStatsAction.java

```diff
@@ -53,6 +65,105 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
+
+        int pageSize = request.paramAsInt("size", 10);  // Default to 10 results per page
+        String nextToken = request.param("next_token");
+        String sortBy = request.param("sort", "node_id"); // Default: sort by node_id
+        String sortOrder = request.param("order", "asc"); // Default: ascending order
+
+        if (!sortOrder.equals("asc") && !sortOrder.equals("desc")) {
+            throw new IllegalArgumentException("Invalid value for 'order'. Allowed: 'asc', 'desc'.");
+        }
+
+        if (!sortBy.equals("node_id") && !sortBy.equals("query_group")) {
+            throw new IllegalArgumentException("Invalid value for 'sort'. Allowed: 'node_id', 'query_group_id'.");
+        }
+
+        boolean isTabular = request.rawPath().contains("_list/wlm_stats");
+
+        if (isTabular) {
+            return channel -> client.admin().cluster().wlmStats(wlmStatsRequest,
+                new RestResponseListener<WlmStatsResponse>(channel) {
+                    @Override
+                    public RestResponse buildResponse(WlmStatsResponse response) throws Exception {
+                        WlmPaginationStrategy paginationStrategy = new WlmPaginationStrategy(pageSize, nextToken, sortBy, sortOrder, response);
+                        List<WlmStats> paginatedStats = paginationStrategy.getPaginatedStats();
+                        Table paginatedTable = buildTable(paginatedStats, paginationStrategy);
+                        PageToken nextPageToken = paginationStrategy.getNextToken();
+
+                        if (nextPageToken != null) {
+                            paginatedTable.setPageToken(nextPageToken);
+                        }
+
+                        request.params().put("v", "true");
+                        return RestTable.buildResponse(paginatedTable, channel);
+                    }
+                }
+            );
+        }
+
         return channel -> client.admin().cluster().wlmStats(wlmStatsRequest, new RestActions.NodesResponseRestListener<>(channel));
     }
+
+    /**
+     * Builds a tabular response with '|' column separators.
+     */
+    private Table buildTable(List<WlmStats> paginatedStats, WlmPaginationStrategy paginationStrategy) {
+        Table table = new Table();
+        table.startHeaders();
+        table.addCell("NODE_ID", "desc:Node ID");
+        table.addCell("|");
+        table.addCell("QUERY_GROUP_ID", "desc:Query Group");
+        table.addCell("|");
+        table.addCell("TOTAL_COMPLETIONS", "desc:Total Completed Queries");
+        table.addCell("|");
+        table.addCell("TOTAL_REJECTIONS", "desc:Total Rejected Queries");
+        table.addCell("|");
+        table.addCell("TOTAL_CANCELLATIONS", "desc:Total Canceled Queries");
+        table.addCell("|");
+        table.addCell("CPU_USAGE", "desc:CPU Usage");
+        table.addCell("|");
+        table.addCell("MEMORY_USAGE", "desc:Memory Usage");
+        table.endHeaders();
```

Comment:
Lets refactor these lines into a method which creates a Table and adds these headers then return the Table instance.